### ISSUE TITLE
StreamAccuracy triggered every experience

### ISSUE THREAD

**Issue by Mattdl** [ContinualAI/avalanche#622]
Following the docs of [StreamAccuracy](https://avalanche-api.continualai.org/autoapi/avalanche/evaluation/metrics/accuracy/?highlight=streamaccuracy#avalanche.evaluation.metrics.accuracy.StreamAccuracy): "At the end of the entire stream of experiences, this plugin metric reports the average accuracy over all patterns seen in all experiences. ".

However, I'm trying out the data incremental example for CoPE #616 , but when only setting the stream accuracy to true it triggers after each experience being processed. This gives a lot of overhead, and following the docs it seems like it should only be reported once. In some situations it might be interesting to have this after each experience (to know performance on the full test stream), but in this case this is infeasible.

TLDR: feature request to define in StreamAccuracy when it should be triggered (after experience or after entire training stream).

**Comment by AndreaCossu** [comment]
Hi @Mattdl ! Thanks for reaching out. The `StreamAccuracy` operates on the evaluation stream and, as you said, should not emit values after each experience (that is the `ExperienceAccuracy` job). This may be a bug.
Can you provide a script to reproduce the behavior?

**Comment by Mattdl** [comment]
Oh nvm I think I found the culprit:

    for experience in scenario.train_stream:
         cl_strategy.train(experience)
        results.append(cl_strategy.eval(scenario.test_stream))

If I put the `cl_strategy.eval` out of the loop I get the StreamAccuracy only once. Is this how it should be implemented or should the  `cl_strategy.eval` be in the loop while the StreamAccuracy should know which phase the strategy is in?

**Comment by AndreaCossu** [comment]
Great! Yes, in your case if you want a single stream accuracy at the end of the entire training, you should place it outside the loop. 

The `StreamAccuracy` knows about the phase in which it is in (train or eval), in fact it does not print any values during the `train` phase. It simply returns a value for each call to `eval`.

**Comment by Mattdl** [comment]
Ok I see, thanks a lot for the quick response! (:
